src/bn_wallets.erl

```diff
@@ -327,11 +328,16 @@ jsonrpc_nonce_param(Param, Address, NonceType, Chain) ->
 ) ->
     {ok, blockchain_txn:txn()} | {error, term()}.
 mk_payment_txn_v2(Payer, PaymentList, Nonce, Chain) ->
-    Payments = [blockchain_payment_v2:new(Payee, Bones) || {Payee, Bones} <- PaymentList],
+    Payments = [mk_payment(Payee, Bones, Max) || {Payee, Bones, Max} <- PaymentList],
     Txn = blockchain_txn_payment_v2:new(Payer, Payments, Nonce),
     TxnFee = blockchain_txn_payment_v2:calculate_fee(Txn, Chain),
     {ok, blockchain_txn_payment_v2:fee(Txn, TxnFee)}.
 
+mk_payment(Payee, undefined, true) ->
+    blockchain_payment_v2:new(Payee, max);
+mk_payment(Payee, Bones, false) ->
+    blockchain_payment_v2:new(Payee, Bones).
```

Comment:
nb, this probably needs some validation work or a better error message but this will probably do the job immediately.